pkg/controllers/gateway/gateway_controller_test.go

```diff
@@ -606,9 +609,9 @@ func TestGatewayReconciler_reconcileTLS(t *testing.T) {
 			r := &GatewayReconciler{
 				Client:       tt.fields.Client,
 				Scheme:       tt.fields.Scheme,
-				Certificates: test.NewTestCertificateService(tt.fields.Client),
-				HostService:  test.NewTestHostService(tt.fields.Client),
-				Placement:    test.NewTestGatewayPlacer(),
+				Certificates: faketls.NewTestCertificateService(tt.fields.Client),
+				HostService:  fakedns.NewTestHostService(tt.fields.Client),
+				Placement:    fakeplacement.NewTestGatewayPlacer(),
```

Comment:
Do we need [this](https://github.com/Kuadrant/multicluster-gateway-controller/blob/3deb7ec77a561be047e05ba150e71cdef89d58b5/pkg/controllers/gateway/gateway_controller_test.go#L746) any more? I'd reuse the Pointer helper function just like [here](https://github.com/Kuadrant/multicluster-gateway-controller/blob/3deb7ec77a561be047e05ba150e71cdef89d58b5/pkg/_internal/policy/policy_test.go#L61)